### PR TITLE
Mirror packages to dotnet-security-partners

### PR DESCRIPTION
**Pull request by mmitche**
During the relese automation, push packages from current release to feed accessible to dotnet-security-partners.

### CODE REVIEW

**Comment by mmitche** [comment]
Test https://dev.azure.com/dnceng/internal/_build/results?buildId=2127246&view=results

**Comment by mmitche** [comment]
Dry run looking good.

File: eng/templates/stages/mirror.yml [hunk 1]
```diff
@@ -112,3 +114,47 @@ stages:
         fi
       workingDirectory: $(Pipeline.Workspace)/$(RepoDir)
       displayName: Mirror and tag
+
+  - job: MirrorPackages
```
Comment (MichaelSimons) [inline eng/templates/stages/mirror.yml:118]: I am not a fan of this being a separate job.  Agent allocation at it's worse can really be problematic and can really be a source of frustration during releases.

File: eng/templates/stages/mirror.yml [hunk 2]
```diff
@@ -112,3 +114,47 @@ stages:
...
+    # means that the 'signed' and 'shipping' artifacts are different in content.
+    - download: ${{ parameters.dotnetStagingPipelineResource }}
+      artifact: ${{ variables.packagesArtifactName }}
+      patterns: ${{ variables.shippingPackages }}/*
+      displayName: Download shipping packages
+
+    # Then, upload these to DSP.
+    - ${{ if parameters.isDryRun }}:
+      # List the packages that would be pushed
+      - script: |
```
Comment (MichaelSimons) [inline eng/templates/stages/mirror.yml:147]: This step should have a name.  It appears as CmdLine which isn't very helpful.

**Comment by MichaelSimons** [comment]
This is somewhat orthogonal to this PR but can we and should be cleanup the following feed - https://dev.azure.com/dotnet-security-partners/dotnet/_artifacts/feed/dotnet6-early-access.  This was a POC that feels like it should be removed.

**Review by MichaelSimons — COMMENTED** [review]
Can

**Comment by mmitche** [comment]
New test looked good. https://dev.azure.com/dnceng/internal/_build/results?buildId=2129580&view=results

File: eng/templates/stages/mirror.yml [hunk 3]
```diff
@@ -13,16 +13,18 @@ parameters:
...
 - stage: Mirror
-  displayName: Mirror branch to DSP
+  displayName: Mirror to DSP
   dependsOn:
   - PreRelease
   - MirrorApproval
 
   jobs:
-  - job: Mirror
+  - job: MirrorSource
```
Comment (MichaelSimons) [inline eng/templates/stages/mirror.yml:27]: After consolidating the jobs, I think the original name was better.
```suggestion
  - job: Mirror
```

**Review by MichaelSimons — APPROVED** [review]

File: eng/templates/stages/mirror.yml (hunk 3 above)
Comment (MichaelSimons) [reply eng/templates/stages/mirror.yml:27]: @mmitche - no sure if you saw this comment before merging.

Comment (mmitche) [reply eng/templates/stages/mirror.yml:27]: ahh missed that. One sec will open a follow up.

**Comment by mmitche** [comment]
https://github.com/dotnet/source-build/pull/3303

**Pull request merged**